source/Calamari.Azure/Integration/Websites/Publishing/ResourceManagerPublishProfileProvider.cs

```diff
@@ -4,71 +4,98 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Calamari.Azure.Integration.Security;
+using Calamari.Azure.Util;
 using Calamari.Commands.Support;
 using Microsoft.Azure;
 using Microsoft.Azure.Management.Resources;
 using Microsoft.Azure.Management.Resources.Models;
 using Microsoft.Azure.Management.WebSites;
 using Microsoft.Rest;
 using Newtonsoft.Json.Linq;
+using Octopus.CoreUtilities.Extensions;
 
 namespace Calamari.Azure.Integration.Websites.Publishing
 {
     public class ResourceManagerPublishProfileProvider
     {
         public static SitePublishProfile GetPublishProperties(string subscriptionId, string resourceGroupName, string siteName, string tenantId, string applicationId, string password,string resourceManagementEndpoint, string activeDirectoryEndPoint)
         {
-            var token = ServicePrincipal.GetAuthorizationToken(tenantId, applicationId, password, resourceManagementEndpoint, activeDirectoryEndPoint);
```

Comment:
Ignore all the changes shown by the PR - the only actual change is wrapping it up int a try/catch block